website/server/controllers/api-v3/groups.js

```diff
@@ -561,6 +561,12 @@ api.joinGroup = {
       }
     }
 
+    if (group.type === 'guild') {
```

Comment:
You will want to read through these examples: https://github.com/HabitRPG/habitica/blob/develop/test/api/v3/integration/groups/POST-groups_groupId_join.test.js

To run test, use `gulp test:api-v3:integration` from the command line.

It is helpful to note that you can run individual tests by adding `it.only`